### PR TITLE
macos_buildenv.sh script improvements

### PR DESCRIPTION
**Pull request by Be-ing**
* Make script runable from ~/.profile
* Check that script is being run with "source" prefix
* Add SHA256 checksum for build environment download
* Fix export of Qt5_DIR
* Don't print exported environment variables; just export them

### CODE REVIEW

**Comment by Be-ing** [comment]
This makes it super easy to setup a Mixxx development environment on macOS:
```
cd
git clone https://github.com/mixxxdj/mixxx.git
echo 'source ~/mixxx/tools/macos_buildenv.sh setup --profile' >> ~/.profile
source ~/mixxx/tools/macos_buildenv.sh setup
cmake -S ~/mixxx -B ~/mixxx/cmake_build
cmake --build ~/mixxx/cmake_build --parallel $(sysctl -n hw.physicalcpu)
```

**Comment by Holzhaus** [comment]
Thanks for taking care! Making it easy to set up the environment without losing flexibility when setting build options (i.e. by *not* wrapping `cmake`) was the idea behind this script. 

> * Don't print exported environment variables; just export them

Can you explain why? When putting this into your `~/.profile` you won't see these messages anyway, and when sourcing it manually wouldn't it be better to provide some kind of feedback what was changed?

**Comment by Be-ing** [comment]
> Can you explain why? When putting this into your ~/.profile you won't see these messages anyway, and when sourcing it manually wouldn't it be better to provide some kind of feedback what was changed?

I found it annoying when running manually. There is now feedback saying the build environment has been found when running manually which I think is sufficient.

**Comment by Be-ing** [comment]
macOS build succeeded. Ready for merge?

File: cmake/macos_build_environment [hunk 1]
```diff
@@ -0,0 +1,2 @@
+2.3-55d9a17-sdk10.15-macosminimum10.12-x86_64
+86c56d3aba24ec23923ad906f5a13084b0f784c955812448ecb43821fd907d50
```
Comment (Holzhaus) [inline cmake/macos_build_environment:2]: Couldn't we just run sha256sum when building the environment and deploy a second file `<buildenvname>.tar.gz.sha256sum` to the server? That's the standard way to do this IMHO. 

Then we can run `sha256sum -c` locally which would simplify the code and we won't have to maintain the checksum in the Mixxx repo.

File: src/library/dao/playlistdao.cpp [hunk 2]
```diff
@@ -458,7 +458,7 @@ void PlaylistDAO::removeTrackFromPlaylist(int playlistId, int position) {
 
 void PlaylistDAO::removeTracksFromPlaylist(int playlistId, const QList<int>& positions) {
     // get positions in reversed order
-    QList sortedPositons = positions;
+    auto sortedPositons = positions;
```
Comment (Holzhaus) [inline src/library/dao/playlistdao.cpp:461]: Unrelated?

**Review by Holzhaus — COMMENTED** [review]
Thank you!

File: cmake/macos_build_environment (hunk 1 above)
Comment (Be-ing) [reply cmake/macos_build_environment:2]: I think it's better to keep this in the code repository. That way we can know what build environment was used for any given build.

File: src/library/dao/playlistdao.cpp (hunk 2 above)
Comment (Be-ing) [reply src/library/dao/playlistdao.cpp:461]: This was necessary to build on macOS 10.13 with an older compiler. I could change it to `QList<int>` if you prefer. Either works.

File: cmake/macos_build_environment (hunk 1 above)
Comment (Be-ing) [reply cmake/macos_build_environment:2]: This also allows us to revert to an older build environment if something goes wrong with a new one.

Comment (Holzhaus) [reply cmake/macos_build_environment:2]: This is just about the sha256 sum. I agree we should keep the buildenv name in the repository.

Comment (Be-ing) [reply cmake/macos_build_environment:2]: If you want to implement that, go ahead. This works fine though.
